packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/ext/RealmResultsExt.kt

```diff
@@ -0,0 +1,104 @@
+ * @param timeout How long to wait for the server to return the objects defined by the subscription.
+ * This is only relevant for [WaitForSync.ALWAYS] and [WaitForSync.FIRST_TIME].
+ * @return The result of running the query against the local Realm file. The results returned will
+ * depend on which [mode] was used.
+ * @throws kotlinx.coroutines.TimeoutCancellationException if the specified timeout was hit before
+ * a query result could be returned.
+ * @Throws IllegalStateException if this method is called on a Realm that isn't using Flexible Sync.
+ */@ExperimentalFlexibleSyncApi
+public suspend fun <T : RealmObject> RealmResults<T>.subscribe(
+    mode: WaitForSync = WaitForSync.FIRST_TIME,
+    timeout: Duration = Duration.INFINITE
+): RealmResults<T> {
+    val query: RealmQuery<T> = this.query("")
+    return query.subscribe(mode, timeout)
+}
+
+/**
+ * Automatically create a named [Subscription] from a local query result in the background and
+ * return the result of re-running the same query against the Realm file.
+ *
+ * This is a more streamlined alternative to doing something like this:
+ *
+ * ```
+ * fun suspend getData(realm: Realm): RealmResults<Person> {
+ *     val results = realm.query<Person>().find()
+ *     realm.subscriptions.update { bgRealm ->
+ *         add("myquery", results.query(""))
+ *     }
+ *     realm.subscriptions.waitForSynchronization()
+ *     return realm.query<Person>().find()
+ * }
+ * ```
+ *
+ * It is possible to define whether or not to wait for the server to send all data before
+ * running the local query. This is relevant as there might be delay from creating a subscription
+ * to the data being available on the device due to either latency or because a large dataset needs
+ * be downloaded.
+ *
+ * The default behaviour is that the first time `subscribe` is called, the query result will not
+ * be returned until data has been downloaded from the server. On subsequent calls to `subscribe`
+ * for the same query, the query will run immediately on the local database while any updates
+ * are downloaded in the background.
+ *
+ * @param name name of the subscription. This can be used to identify it later in the [SubscriptionSet].
+ * @param mode type of mode used to resolve the subscription. See [WaitForSync] for more details.
+ * @param timeout How long to wait for the server to return the objects defined by the subscription.
+ * This is only relevant for [WaitForSync.ALWAYS] and [WaitForSync.FIRST_TIME].
+ * @return The result of running the query against the local Realm file. The results returned will
+ * depend on which [mode] was used.
+ * @throws kotlinx.coroutines.TimeoutCancellationException if the specified timeout was hit before
+ * a query result could be returned.
+ * @Throws IllegalStateException if this method is called on a Realm that isn't using Flexible Sync.
+ */
+@ExperimentalFlexibleSyncApi
+public suspend fun <T : RealmObject> RealmResults<T>.subscribe(
+    name: String,
```

Comment:
Would it make sense to collapse the anonymous and named variant by allowing `name: String? = null` to reduce the API surface 🤔

Reply:
I remember why I split it, the reason was two-fold:

1) `name` and `updateExisting` kinda goes hand-in-hand. With all named parameters, you could set `name = null` and `updateExisting = true`, which wouldn't make sense.

2) We assume the primary use case is going to be the anonymous one and setting the mode, with all optional parameters is going to be slightly awkward. I guess we could move `mode/timeout` to the front of the parameters. But that felt a bit "off" to me. 

Not sure if that is argument enough to not collapse them though 🤔 The methods are marked experimental so we can change them later.